filters/ratelimit/ratelimit_test.go

```diff
@@ -439,3 +467,31 @@ func TestAllowsContext(t *testing.T) {
 		t.Errorf("unexpected response: %v", ctx.FResponse)
 	}
 }
+
+func TestGetKeyShards(t *testing.T) {
+	for _, tc := range []struct {
+		maxHits      int
+		maxKeyShards int
+		want         int
+	}{
+		{1, 0, 1},
+		{1, 1, 1},
+		{100, 1, 1},
+		{4, 5, 4},
+		{5, 5, 5},
+		{6, 5, 3},
+		{11, 10, 1}, // maxHits is prime
+		{12, 10, 6},
```

Comment:
My bet is that if we have RR for N skipper to M redis, maxHitsPerShard is small and time.Duration is small it will be too blurry. Example
10 skipper and 10 redis with maxHits=10 and maxHitsPerShard=1 and 1s time interval.
My expectation is if we have maxHitsPerShard within 100-1000 it would work better than with low numbers like <10.
Maybe I am wrong.